ios/browser/api/brave_shields/adblock_service.mm

```diff
@@ -93,4 +93,20 @@ - (void)unregisterFilterListComponent:(AdblockFilterListCatalogEntry*)entry {
   _cus->UnregisterComponent(base::SysNSStringToUTF8(entry.iosComponentId));
 }
 
+- (void)registerComponentForComponentInfo:(ComponentRegistrationInfo*)info
+                           componentReady:(void (^)(NSString* _Nullable installPath)) componentReady {
+  brave_shields::RegisterAdBlockFiltersComponent(
+      _cus, base::SysNSStringToUTF8(info.base64PublicKey),
+      base::SysNSStringToUTF8(info.componentId),
+      base::SysNSStringToUTF8(info.title),
+      base::BindRepeating(^(const base::FilePath& install_path) {
+        const auto installPath = base::SysUTF8ToNSString(install_path.value());
+        componentReady(installPath);
```

Comment:
How do we generate `install_path` here? Do we use filter's name to generate it?

Reply:
@stoletheminerals  
To answer your question it came from the component updated (that part I didn't change if you look it closely).
But my approach has changed. 

I wanted to pass raw data but decided to put back the old function as it was and just pass a boolean legacy or not so the changes are much simpler now

Reply:
@cuba my concern was that `installPath` could be used to overwrite arbitrary files. Do you have an example for this value? If it comes from the component updater I assume it has a CRX ID for each component, but wanted to double check.

Reply:
@stoletheminerals The install path comes from component updater: https://github.com/brave/brave-core/blob/539a4c374f3667c8d00fe9c807b396947a39a3ad/components/brave_shields/browser/ad_block_component_installer.cc#L185